src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -191,7 +191,7 @@ public ValidationContext IntegrityValidation(ChainedHeader header, Block block)
         }
 
         /// <inheritdoc/>
-        public async Task<ValidationContext> FullValidationAsync(ChainedHeader header, Block block)
+        public virtual async Task<ValidationContext> FullValidationAsync(ChainedHeader header, Block block)
```

Comment:
Can you also add virtual to the other validation methods?

Reply:
fixed